File: sdk/swift/Sources/AlignTechFeatureSDK/ConfigStore.swift
import Foundation

actor ConfigStore {
    private var flags: [String: FlagConfig] = [:]
    private var version: String = ""
    
    func replaceAll(version: String, flags: [String: FlagConfig]) {
        self.version = version
        self.flags = flags
    }
    
    func get(_ flagKey: String) -> FlagConfig? {
        return flags[flagKey]
    }
    
    func getVersion() -> String {
        return version
    }
    
    func getSize() -> Int {
        return flags.count
    }
    
    func isEmpty() -> Bool {
        return flags.isEmpty
    }
}

File: sdk/dotnet/src/AlignTech.FeatureSDK/ConfigStore.cs
using System.Collections.Concurrent;
using AlignTech.FeatureSDK.Models;

namespace AlignTech.FeatureSDK;

internal class ConfigStore
{
    private readonly ConcurrentDictionary<string, FlagConfig> _flags = new();

    public void UpdateFlags(IEnumerable<FlagConfig> flags)
    {
        _flags.Clear();
        foreach (var flag in flags)
        {
            _flags[flag.FlagKey] = flag;
        }
    }

    public FlagConfig? Get(string flagKey)
    {
        _flags.TryGetValue(flagKey, out var flag);
        return flag;
    }

    public bool IsEmpty() => _flags.IsEmpty;
}
